tests/zfs-tests/tests/functional/inheritance/inherit_001_pos.ksh

```diff
@@ -331,7 +331,7 @@ function scan_state { #state-file
 				if [[ $op == "-" ]]; then
 					log_note "No operation specified"
 				else
-					export __ZFS_POOL_RESTRICT="$TESTPOOL"
+					export __ZFS_POOL_RESTRICT="TESTPOOL"
```

Comment:
At Delphix, we have the change you describe above (using "TESTPOOL") and the test passes. We haven't yet upstreamed the change that introduced the fix to OpenZFS yet though.